teuthology/task/ansible.py

```diff
@@ -177,7 +180,11 @@ def generate_hosts_file(self):
         hosts = self.cluster.remotes.keys()
         hostnames = [remote.hostname for remote in hosts]
         hostnames.sort()
-        hosts_str = '\n'.join(hostnames + [''])
+        inventory = []
+        if self.inventory_group:
+            inventory.extend(['[{0}]'.format(self.inventory_group)])
```

Comment:
This might be refactoring cruft. It would be clearer as ``list.append()`` here as it's the first thing going in the list, no?